c2rust-transpile/src/translator/variadic.rs

```diff
@@ -9,130 +9,110 @@ pub enum VaPart {
-            expr: CExprId,
-        ) -> Option<CDeclId> {
-            match_or! { [ast_context[expr].kind]
-            CExprKind::ImplicitCast(_, me, _, _, _) => me }
-            match_or! { [ast_context[me].kind]
-            CExprKind::Member(_, e, _, _, _) => e }
-            match_or! { [ast_context[e].kind]
-            CExprKind::DeclRef(_, va_id, _) => va_id }
+        // `struct`-based `va_list` (e.g. x86_64) where `va_list` is accessed as a `struct` member.
+        //
+        // Supporting this pattern is necessary to transpile apache httpd.
+        let match_vastart_struct_member = || {
+            let me = match_or! { [ast[expr].kind]
+            CExprKind::ImplicitCast(_, me, _, _, _) => me }?;
+            let e = match_or! { [ast[me].kind]
+            CExprKind::Member(_, e, _, _, _) => e }?;
+            let va_id = match_or! { [ast[e].kind]
+            CExprKind::DeclRef(_, va_id, _) => va_id }?;
             Some(va_id)
-        }
+        };
 
-        // struct-based va_list (e.g. x86_64) where va_list is accessed as a member of a struct pointer
-        // supporting this pattern is necessary to transpile [graphviz](https://gitlab.com/graphviz/graphviz/-/blob/5.0.0/lib/sfio/sftable.c#L321)
-        fn match_vastart_struct_pointer_member(
-            ast_context: &TypedAstContext,
-            expr: CExprId,
-        ) -> Option<CDeclId> {
-            match_or! { [ast_context[expr].kind]
-            CExprKind::ImplicitCast(_, me, _, _, _) => me }
-            match_or! { [ast_context[me].kind]
-            CExprKind::Member(_, ie, _, _, _) => ie }
-            match_or! { [ast_context[ie].kind]
-            CExprKind::ImplicitCast(_, e, _, _, _) => e }
-            match_or! { [ast_context[e].kind]
-            CExprKind::DeclRef(_, va_id, _) => va_id }
+        // `struct`-based `va_list` (e.g. x86_64) where `va_list` is accessed as a member of a `struct *` pointer.
+        //
+        // Supporting this pattern is necessary to transpile
+        // [graphviz](https://gitlab.com/graphviz/graphviz/-/blob/5.0.0/lib/sfio/sftable.c#L321).
+        let match_vastart_struct_pointer_member = || {
+            let me = match_or! { [ast[expr].kind]
+            CExprKind::ImplicitCast(_, me, _, _, _) => me }?;
+            let ie = match_or! { [ast[me].kind]
+            CExprKind::Member(_, ie, _, _, _) => ie }?;
+            let e = match_or! { [ast[ie].kind]
+            CExprKind::ImplicitCast(_, e, _, _, _) => e }?;
+            let va_id = match_or! { [ast[e].kind]
+            CExprKind::DeclRef(_, va_id, _) => va_id }?;
             Some(va_id)
-        }
+        };
 
-        // char pointer-based va_list (e.g. x86)
-        fn match_vastart_pointer(ast_context: &TypedAstContext, expr: CExprId) -> Option<CDeclId> {
-            match_or! { [ast_context[expr].kind]
-            CExprKind::DeclRef(_, va_id, _) => va_id }
+        // `char *` pointer-based `va_list` (e.g. x86).
```

Comment:
Same here, pick one of "`char *`" or "`char` pointer", to match convention in other comment.

Reply:
Fixed in https://github.com/immunant/c2rust/pull/638/commits/8fae26f66fa8e21116cb3f8c8e646a50846152de.